### PR TITLE
Rework and simplify our configuration.

### PR DESCRIPTION
**Pull request by kopepasah**
This pull request reworks the configuration, cleans up the setup test, and adds additional tests for verification.

- [x] Rework action config to only output the required override file.
- [x] Add initial tests for validating start, and checking WordPress and PHP versions.
- [x] Update README.md with general information and usage instructions.
- [x] Add a CODEOWNERS file.

Post review updates:

- [x] Remove CODEOWNERS file.
- [x] Fix comment to add `.` to `.wp-env.json` and change "error" to "warning".

### CODE REVIEW

**Comment by kopepasah** [comment]
Once this is merged, I will create the initial release (`v1`) and `1.0.0` tag.

**Comment by kopepasah** [comment]
@olafleur and @jrtashjian please have a look at this pull request and let me know what you think. Once approved, I will merge, create our initial releases, then publish to the Github Actions Marketplace.

File: .github/CODEOWNERS
Comment (olafleur-godaddy) [inline .github/CODEOWNERS]: We should probably discuss this, as we removed the CODEOWNERS file in the past.

File: .github/workflows/test-wordpress-version.yml
Comment (olafleur-godaddy) [inline .github/workflows/test-wordpress-version.yml:9]: We want to test WP 4 ?

File: .github/workflows/test-wordpress-version.yml
Comment (olafleur-godaddy) [inline .github/workflows/test-wordpress-version.yml:29]: And 5.4 ? What is the logic behind the choosing of these versions?

File: README.md
Comment (olafleur-godaddy) [inline README.md]: best practice TO use

File: README.md
Comment (olafleur-godaddy) [inline README.md]: Maybe we should replace with (or add) https://twitter.com/GoDaddyOSS, which is maybe more relevant to devs in the OSS space.

File: action.yml
Comment (olafleur-godaddy) [inline action.yml]: an -> a

File: action.yml
Comment (olafleur-godaddy) [inline action.yml:39]: I don't think a missing `.wp-env.override.json` file will throw an error. You could use only the `.wp-env.json` if you wanted.

**Review by olafleur-godaddy — APPROVED** [review]
Looks Good To Me! :)

File: .github/CODEOWNERS
Comment (kopepasah) [reply .github/CODEOWNERS]: Okay, that makes sense. My opinion would be that for OSS repos, someone needs to own the code. Regardless if we have an official CODEOWNERS file or not, having someone as a stakeholder ensures the software does not go stale.

File: .github/workflows/test-wordpress-version.yml
Comment (kopepasah) [reply .github/workflows/test-wordpress-version.yml:9]: We are not testing WP 4, but rather that the core version passed in the config installs the correct WP version. Since `@wordpress/env` allows installing any version, we ensure here that the version passed in the Action config is the one actually installed.

File: .github/workflows/test-wordpress-version.yml
Comment (kopepasah) [reply .github/workflows/test-wordpress-version.yml:29]: Same answer, regarding testing the Action config.

File: action.yml
Comment (kopepasah) [reply action.yml:39]: This Action assumes no `.wp-env.json` is present, although it is possible to include one in the project. We likely need to add some sort of documentation in the README.md that talks about the usage of a `wp-env.json` within the repo, and how this overrides that config.

Otherwise, if the project that uses this action does _not_ have a `wp-env.json`, this will fail, which is why we have that note. Let me consider how we might rework, as well as add some additional documentation to the README.md.

File: README.md
Comment (kopepasah) [reply README.md]: Good call. I pulled this from Go, but this makes sense. Fixed.

File: action.yml
Comment (kopepasah) [reply action.yml]: Fixed. 😃

File: action.yml
Comment (kopepasah) [reply action.yml:39]: I added some information regarding `wp-env.json` to the README.md.

File: README.md
Comment (kopepasah) [reply README.md]: Fixed.

File: .github/CODEOWNERS
Comment (olafleur-godaddy) [reply .github/CODEOWNERS]: Agreed. When we had a CODEOWNERS file, we had almost all the devs in it, I think ;)

File: README.md
Comment (olafleur-godaddy) [inline README.md]: `.wp-env.json` (You are missing a dot at the beginning)

File: action.yml
Comment (jrtashjian) [reply action.yml:39]: @kopepasah I'm surprised this fails if the `.wp-env.json` file is missing. Was there an error describing why?

File: .github/CODEOWNERS
Comment (jrtashjian) [reply .github/CODEOWNERS]: Another reason was the review reminder notifications in Slack. We got lazy assigning the proper people so everyone was pinged for everything and the result was slacking on reviews and stories dropping.

I think this is fine to keep if you want since it's a small project and this kind of stuff is where you are wanting to be focused.

**Review by jrtashjian — APPROVED** [review]
This all looks great!

File: .github/CODEOWNERS
Comment (kopepasah) [reply .github/CODEOWNERS]: Thanks for the feedback! I will remove the file, actually. Unless we come up with a solution as a team to use this file, let's keep it out of our repos.

File: README.md
Comment (kopepasah) [reply README.md]: Going to update this comment by adding the `.` and also changing the "error" to warning (because that's what @wordpress/env throws), although the environment does start.

File: action.yml
Comment (olafleur-godaddy) [inline action.yml]: a warning (not an)

Comment (kopepasah) [reply action.yml]: Fixed.

**Pull request merged**
